packages/react-components/react-utilities/src/trigger/getTriggerChild.ts

```diff
@@ -19,11 +20,23 @@ import { isFluentTrigger } from './isFluentTrigger';
  *   </Tooltip>
  * );
  * ```
+ *
+ * In the case where the immediate child is not a valid element,
+ * null is returned
  */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const getTriggerChild = <P = any>(
-  children: React.ReactNode,
-): React.ReactElement<P> & { ref?: React.Ref<unknown> } => {
-  const child = React.Children.only(children) as React.ReactElement;
-  return isFluentTrigger(child) ? getTriggerChild(child.props.children) : child;
-};
+export function getTriggerChild<TriggerChildProps>(
+  children: TriggerProps<TriggerChildProps>['children'],
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): (React.ReactElement<Partial<TriggerChildProps>> & { ref?: React.Ref<any> }) | null {
+  if (!React.isValidElement<TriggerChildProps>(children)) {
+    return null;
+  }
+  const child = React.Children.only(children);
```

Comment:
```suggestion
```

I don't see a reason to use there `React.Children.only()` as you have already done a check for `React.isValidElement`.